common/foreign.yaml

```
@@ -0,0 +1,26 @@
+datasets:
+- data_type_def: ForeignFieldTable
+  doc: A table for identifying which objects in a file exist
+       outside the file
+  dtype:
+   - name: id
+     dtype: uint
```

Minor suggestion: specify minimum precision explicitly

```suggestion
     dtype: uint32
```

---

Also for the uints in `OntologyTable` below.